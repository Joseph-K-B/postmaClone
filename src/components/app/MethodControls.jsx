import React from 'react';
import css from './styles/controlStyles.css'

function MethodControls({
    url,
    inputField,
    onUrlInput,
    onRadioInput,
    onObjectInput,
    onSubmit
}) {
    return(
        <form onSubmit={onSubmit} className='form'>
            <label htmlFor='url'>Make an API call</label>
            <input 
                id='url'
                name='url'
                type='text'
                value={url}
                onChange={onUrlInput}
                className='url-input'
            />
            <div className='radio-div'>
            <input 
                type='radio' 
                id='GET' 
                name='GET' 
                value='GET' 
                onChange={onRadioInput}
                className='radio-input'
            />
            <label htmlFor='GET'>GET</label>
            <input 
                type='radio' 
                id='POST' 
                name='POST' 
                value='POST' 
                onChange={onRadioInput}
                className='radio-input'
                />
            <label htmlFor='POST'>POST</label>
            <input 
                type='radio' 
                id='PUT' 
                name='PUT' 
                value='PUT' 
                onChange={onRadioInput}
                className='radio-input'
                />
            <label htmlFor='PUT'>PUT</label>
            <input 
                type='radio' 
                id='PATCH' 
                name='PATCH' 
                value='PATCH' 
                onChange={onRadioInput}
                className='radio-input'
            />
            <label htmlFor='PATCH'>PATCH</label>
            <input 
                type='radio' 
                id='DELETE' 
                name='DELETE' 
                value='DELETE'
                onChange={onRadioInput}
                className='radio-input'
            />
            <label htmlFor='DELETE'>DELETE</label>
            </div>
            <button aria-label='make-request'>Go</button>
            <label htmlFor= 'object'></label>
            <textarea
                id='object'
                name='object'
                type='text'
                value={inputField}
                onChange={onObjectInput}
                className='object-input'
                placeholder='Double quotes for keys & values'
            ></textarea>
        </form>
    )
}

export default MethodControls