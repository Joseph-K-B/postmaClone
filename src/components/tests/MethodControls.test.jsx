import React from 'react';
import { render } from '@testing-library/react';
import MethodControls from '../app/MethodControls';

describe('renders form/controls', () => {
    it('should display all inputs and button', () => {
        render(
            <MethodControls />
        );
        expect(MethodControls).toMatchSnapshot();
    });
});