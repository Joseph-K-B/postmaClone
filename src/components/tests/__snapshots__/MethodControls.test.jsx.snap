// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form/controls should display all inputs and button 1`] = `[Function]`;
